runtime/caml/sizeclasses.h

```diff
@@ -1,19 +1,19 @@
 /* This file is generated by tools/gen_sizeclasses.ml */
 #define POOL_WSIZE 4096
-#define POOL_HEADER_WSIZE 4
+#define POOL_HEADER_WSIZE 32
 #define SIZECLASS_MAX 128
 #define NUM_SIZECLASSES 32
 static const unsigned int wsize_sizeclass[NUM_SIZECLASSES] = 
-{ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 12, 14, 16, 18, 20, 23, 26, 29, 33, 37, 42,
-  47, 53, 59, 65, 73, 81, 89, 99, 108, 118, 128 };
+{ 1, 2, 3, 4, 5, 6, 7, 8, 10, 12, 14, 16, 17, 19, 22, 25, 28, 32, 34, 38, 43,
+  48, 53, 59, 66, 73, 81, 89, 98, 107, 117, 128 };
```

Comment:
The use of cache-line-alignment I know involves atomic arrays, used to spread a load among non-interfering atomic values. To have cache-line-aligned atomic arrays, one would need all multiples of 16. Is having cache-line-aligned atomic arrays something you consider necessary eventually?

Reply:
We have some structures in lockfree, which use arrays of atomics. It would be useful for better caching. But I think it's a bigger lift because atomics are not immediate now (so an array `'a Atomic.t array`, where atomics have been instantiated with `make_aligned` should not experience false-sharing as far as atomic operations are concerned). 

Agreed about having all multiples of 16 there, and perhaps also a special path for large allocations. Either that or a nice wrapper to allocate a bigger array and rely on padding, I suppose.

(this code changed since header size is at 4 byte again)